File: code_examples/Multi-Platform/src/c/apb.c
void apb(int nx, double *a, double *b, double *c) {
	// size and intent of array arguments for f2py
	// a :: nx, in
	// b :: nx, in
	// c :: nx, inout
	
	int i;

	for (i=0; i<nx; i++) {
		c[i] = KK*a[i] + bpc(b[i], c[i]);
	}
}

File: code_examples/Multi-Platform/src/f90/apb.f90
SUBROUTINE apb(nx, a, b, c)
  USE apb_ext, ONLY : bpc

  IMPLICIT NONE
  INTEGER, INTENT(IN) :: nx
  REAL(8), DIMENSION(nx), INTENT(IN) :: a, b
  REAL(8), DIMENSION(nx), INTENT(INOUT) :: c

  INTEGER :: i

  DO i=1,nx
    c(i) = KK*a(i) + bpc(b(i), c(i))
  END DO
END SUBROUTINE apb
